pkg/svcat/service-catalog/plan.go

```diff
@@ -33,12 +33,22 @@ const (
 )
 
 // RetrievePlans lists all plans defined in the cluster.
-func (sdk *SDK) RetrievePlans() ([]v1beta1.ClusterServicePlan, error) {
+func (sdk *SDK) RetrievePlans(opts *FilterOptions) ([]v1beta1.ClusterServicePlan, error) {
 	plans, err := sdk.ServiceCatalog().ClusterServicePlans().List(v1.ListOptions{})
 	if err != nil {
 		return nil, fmt.Errorf("unable to list plans (%s)", err)
 	}
 
+	if opts != nil && opts.ClassID != "" {
```

Comment:
When class id is set you can use this existing function to perform a filtered list through the API server, instead of filtering on our end.

https://github.com/kubernetes-incubator/service-catalog/blob/ec0548639595f237a87faa4ada366b47cb49108e/pkg/svcat/service-catalog/plan.go#L73-L84